discord/poll.py

```diff
@@ -0,0 +1,644 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+
+from typing import (
+    Dict,
+    Optional,
+    List,
+    TYPE_CHECKING,
+    Union,
+    AsyncIterator,
+)
+import datetime
+
+from .enums import PollLayoutType, try_enum
+from . import utils
+from .emoji import PartialEmoji, Emoji
+from .user import User
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .message import Message, PartialMessage
+    from .abc import Snowflake
+    from .state import ConnectionState
+    from .member import Member
+
+    from .types.poll import (
+        Poll as PollPayload,
+        PollMedia as PollMediaPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollWithExpiry as PollWithExpiryPayload,
+        FullPoll as FullPollPayload,
+        PollAnswerWithID as PollAnswerWithIDPayload,
+    )
+
+    PollMessage = Union[PartialMessage, Message]
+
+
+__all__ = (
+    'Poll',
+    'PollAnswer',
+    'PollAnswerCount',
+    'PollMedia',
+)
+
+MISSING = utils.MISSING
+
+# PollDuration = Literal[1, 4, 8, 24, 72, 168]
+# So, I discovered that any time is valid, so there is no need for this to be a strict
+# literal.
+
+
+PollMediaEmoji = Union[PartialEmoji, Emoji, str]
+
+
+
+class PollMedia:
```

Comment:
I think this could be a NamedTuple subclass?

Reply:
I thought of it when I first created it, but I didn't subclass it at the end.
Going to change though.

Reply:
You are right, this is now a NamedTuple.